scipy/integrate/quadpack.py

```diff
@@ -363,8 +363,40 @@ def quad(func, a, b, args=(), full_output=0, epsabs=1.49e-8, epsrel=1.49e-8,
         else:
             warnings.warn(msg, IntegrationWarning)
             return retval[:-1]
-    else:
-        raise ValueError(msg)
+
+    elif ier == 6:  # Forensic decision tree when QUADPACK throws ier=6
+        if epsabs <= 0:  # Small error tolerance - applies to all methods
+            if epsrel <= 50 * sys.float_info.epsilon or epsrel < 5e-29:
+                msg =- "If ``errabs<=0``, `epsrel` must be greater than both 5e-29 and 50*(machine epsilon)."
+            elif weight in ['sin', 'cos'] and (abs(a) + abs(b) == Inf):
+                msg = "sin- or cos- weighted intergals with infinite bounds must have ``epsabs>0``."
+
+        elif weight is None:
+            if points is None:  # QAGSE/QAGIE
+                msg = "Number of subintervals `limit` must be >=1."
```

Comment:
This sentence doesn't read correctly. Is `limit` word part of the numer of subintervals?

Reply:
Wasn't sure if I should refer to the variable name (limit) or its description (# of subintervals).  The requirement is `limit>=1`.  How would you phrase it?

Reply:
Ah ok, now I get it. If `limits` is clear from the context, I would say, `Invalid 'limit' argument. There must be at least one subinterval` .